parsely/src/main/java/com/parsely/parselyandroid/EngagementManager.kt

```diff
@@ -0,0 +1,89 @@
+package com.parsely.parselyandroid
+
+import kotlin.time.Duration
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+
+/**
+ * Engagement manager for article and video engagement.
+ *
+ *
+ * Implemented to handle its own queuing of future executions to accomplish
+ * two things:
+ *
+ *
+ * 1. Flushing any engaged time before canceling.
+ * 2. Progressive backoff for long engagements to save data.
+ */
+internal class EngagementManager(
+    private val parselyTracker: ParselyTracker,
+    private var latestDelayMillis: Long,
+    private val baseEvent: Map<String, Any>,
+    private val intervalCalculator: HeartbeatIntervalCalculator,
+    private val coroutineScope: CoroutineScope,
+    private val clock: Clock,
+) {
+    private var job: Job? = null
+    private var totalTime: Long = 0
+    private var startTime: Duration
```

Comment:
I think we can use a local variable for this.

Since we are setting it at the beginning of `start()`, I don't think we need to do it in `init`. Once that's removed, we'll only be using it in `start()` and can be converted to a local variable.

Reply:
Right, good point, thanks! Addressed in d282f8b1117103d251ce3dc2a213b3a27c72f3b3